### DISCUSSION TITLE
'shape' or 'element' like method for record()

### DISCUSSION THREAD

**Discussion by medihack** [colinhacks/zod#655] [Q&A]
Is there a way to get the internals of a record? Something like `shape` for `object` or `element` for `array`?
For example I want to get `ZodString` from `schema` of `const schema = z.record(z.string())`

**Comment by scotttrinh** [comment]
Good question. There is a `_def.valueType` that works, but I'm not sure if there is a public interface similar to `shape`.

```ts
import test from "tape";
import { z } from "zod";

const schema = z.record(z.string());

test("valid", (t) => {
  t.true(schema._def.valueType instanceof z.ZodString);
  t.end();
});
```
[CodeSandbox](https://codesandbox.io/s/snowy-moon-2g7yf?file=/src/index.ts:0-183)

**Comment by medihack** [comment]
Thanks, that works. Still, I think there should be a public API for that. I'll create an issue.
